openproblems/tasks/spatial_decomposition/methods/cell2location.py

```diff
@@ -0,0 +1,112 @@
+from ....tools.decorators import method
+from ....tools.utils import check_version
+from ..utils import split_sc_and_sp
+
+import numpy as np
+
+
+@method(
+    method_name="Cell2location",
+    paper_name="Cell2location maps fine-grained cell types in spatial transcriptomics",
+    paper_url="https://www.nature.com/articles/s41587-021-01139-4",
+    paper_year=2022,
+    code_url="https://github.com/BayraktarLab/cell2location",
+    image="openproblems-python-extras",
+)
+def stereoscope(adata, test=False, max_epochs_sc=None, max_epochs_sp=None):
+
+    from cell2location.models import Cell2location
+    from cell2location.models import RegressionModel
+
+    if test:
+        max_epochs_sp = max_epochs_sp or 10
+        max_epochs_sc = max_epochs_sc or 10
+        num_samples = 10
+    else:  # pragma: nocover
+        max_epochs_sc = max_epochs_sc or 250
+        max_epochs_sp = max_epochs_sp or 30000
+        num_samples = 1000
+
+    adata_sc, adata = split_sc_and_sp(adata)
+
+    if "tech" in adata_sc.obs.columns:
+        adata_sc.obs["batch_key"] = adata_sc.obs["tech"].copy()
+    else:
+        adata_sc.obs["batch_key"] = "all"
+
+    # REFERENCE SIGNATURE ESTIMATION FROM SCRNA
+    # prepare anndata for the regression model
+    RegressionModel.setup_anndata(
+        adata=adata_sc,
+        # 10X reaction / sample / batch
+        batch_key="batch_key",
+        # cell type, covariate used for constructing signatures
+        labels_key="label",
+    )
+    sc_model = RegressionModel(adata_sc)
+    sc_model.train(max_epochs=max_epochs_sc)
+    # In this section, we export the estimated cell abundance
+    # (summary of the posterior distribution).
+    adata_sc = sc_model.export_posterior(
+        adata_sc, sample_kwargs={"num_samples": num_samples, "batch_size": 2500}
```

Comment:
Can you please set `num_samples` and `batch_size` as kwargs as you do with `max_epochs`?

Reply:
See my changes

Reply:
Are the new changes matching what you expect?

Reply:
Perfect!